### DISCUSSION TITLE
How to manage and clear Portal's recent file history?

### DISCUSSION THREAD

**Discussion by rideandrain** [flatpak/xdg-desktop-portal#1228] [Questions] [answered]
How do I manage and clear Portal's recent file history? It appears to keep its own file history separate from DE apps (Dolphin) and does not follow system settings on not keeping file history. Its the same situation for directory history.

I'm on up-to-date Fedora 38, using Plasma KDE.

I posted a [similar question on Fedora's forum](https://discussion.fedoraproject.org/t/how-to-clear-and-manage-recent-file-history-in-portal-related-to-freedesktop-and-flatpak/97048) but with no resolution yet. Any help is appreciated, thank you.

**Comment by GeorgesStavracas** [comment]
This is a missing feature: https://github.com/flatpak/xdg-desktop-portal/issues/215

Any help in this front would be appreciated :slightly_smiling_face:

**Reply by rideandrain** [reply]
I see, thanks for the reply.

I'm not a developer so I can't really help there. I would imagine that these lists are being kept in some file that I could manually delete, that would be a sufficient workaround for me.

**Reply by GeorgesStavracas** [reply]
At least in GNOME, recent files can be cleaned in the "Privacy" tab of the Setting app. This removed the `~/.local/share/recently-used.xbel` file.

I'm afraid I don't know about KDE.

(Don't forget to mark one of the answer as the answer to this question if they help you.)

**Comment by NICHOLAS85** [comment, marked as answer]
I was able to find where this history was located in KDE. At `~/.config/xdg-desktop-portal-kderc` there is a Recent Files= and Recent URLs= entry which you can clear. I found that you can just delete the file and it'll be regenerated on its own.

**Reply by rideandrain** [reply]
Thank you so much for this. I can verify that on KDE, `~/.config/xdg-desktop-portal-kderc` keeps track of recent file history, and that deleting this file does clear my recent file history list.

Recent directories however seem to be stored separately, would you happen to know where it is kept?

**Reply by rideandrain** [reply]
I did some digging and managed to find out where directory history is kept as well!

In the file `~/.config/kdeglobals` under `[DirSelect Dialog]`, clear everything after `History Items[$e]=`

**Reply by rideandrain** [reply]
Updates for Plasma 6.x, the locations have been moved to:
`/home/<username>/.local/state/xdg-desktop-portal-kdestaterc`
`/home/<username>/.local/state/kdirselectstaterc`
